circleguard/visualizer.py

```diff
@@ -381,23 +457,70 @@ def draw_sliderbody(self, painter, hitobj):
+        sliderbody_alpha = 75 - ((self.get_hit_time(hitobj) - current_time - (self.preempt - self.fade_in)) / self.fade_in) * 75
+        fade_out = max(0, ((current_time - self.get_hit_endtime(hitobj)) / self.hitwindow * 0.5))
+        magic = (75 * ((fade_out) * 2))
         sliderbody_alpha = sliderbody_alpha if sliderbody_alpha < 75 else 75
-        c = painter.pen().color()
+        sliderbody_alpha = sliderbody_alpha - (magic if magic > 0 else 0)
+        sliderbody_alpha = sliderbody_alpha if sliderbody_alpha > 0 else 0
+        c = self.painter.pen().color()
 
-        _pen = painter.pen()
-        _pen.setWidth(sliderbody_radius*2+WIDTH_CIRCLE_BORDER)
+        _pen = self.painter.pen()
+        _pen.setWidth(self.hitcircle_radius * 2 + WIDTH_CIRCLE_BORDER)
         _pen.setCapStyle(Qt.RoundCap)
         _pen.setJoinStyle(Qt.RoundJoin)
         _pen.setColor(QColor(c.red(), c.green(), c.blue(), sliderbody_alpha))
 
-        sliderbody.moveTo(hitobj.x+X_OFFSET, hitobj.y+Y_OFFSET)
-        for i in hitobj.curve_points:
-            sliderbody.lineTo(i.x+X_OFFSET, i.y+Y_OFFSET)
+        p = hitobj.position
+        sliderbody.moveTo(p.x + X_OFFSET, p.y + Y_OFFSET)
+        for i in hitobj.slider_body:
+            sliderbody.lineTo(i.x + X_OFFSET, i.y + Y_OFFSET)
+
+        self.painter.setPen(_pen)
+        self.painter.drawPath(sliderbody)
 
-        painter.setPen(_pen)
-        painter.drawPath(sliderbody)
+    def draw_progressbar(self, percentage):
+        loading_bg = QPainterPath()
+        loading_bar = QPainterPath()
+        c = self.painter.pen().color()
+
+        _pen = self.painter.pen()
+        _pen.setWidth(5)
+        _pen.setCapStyle(Qt.RoundCap)
+        _pen.setJoinStyle(Qt.RoundJoin)
+        _pen.setColor(QColor(c.red(), c.green(), c.blue(), 25))
+        self.painter.setPen(_pen)
+
+        loading_bg.moveTo(SCREEN_WIDTH/2 - 75, SCREEN_HEIGHT/2)
+        loading_bg.lineTo(SCREEN_WIDTH/2 - 75 + 150, SCREEN_HEIGHT/2)
+
+        loading_bar.moveTo(SCREEN_WIDTH/2 - 75, SCREEN_HEIGHT/2)
+        loading_bar.lineTo(SCREEN_WIDTH/2 - 75 + percentage * 1.5, SCREEN_HEIGHT/2)
+
+        self.painter.drawPath(loading_bg)
+        _pen.setColor(QColor(c.red(), c.green(), c.blue(), 255))
+        self.painter.setPen(_pen)
+        self.painter.drawPath(loading_bar)
+
+    def draw_loading_screen(self):
+        self.painter.drawText(SCREEN_WIDTH/2 - 75, SCREEN_HEIGHT/2 - 10, f"Calculating Sliders, please wait...")
+        self.draw_progressbar(int((self.sliders_current / self.sliders_total) * 100))
+
+    def proccess_sliders(self):
+        self.sliders_total = len(self.beatmap.hit_objects) - 1
+        for index in range(len(self.beatmap.hit_objects)):
+            self.sliders_current = index
+            current_hitobj = self.beatmap.hit_objects[index]
+            if isinstance(current_hitobj, Slider):
+                try:
```

Comment:
why try/catch? can't you check the other types (Bezier/MultiBezier) first, then in the final else you know it's just a normal slider?